[pickers] Deprecate the `disableOpenPicker` prop (#17040)
mui/mui-x: docs/data/migration/migration-pickers-v7/migration-pickers-v7.md

1. The user cleans the year, the rendered value is `01/01/YYYY`, `onChange` is fired with `null`.
2. The user enters a new year, the rendered value is `01/01/2026`, `onChange` is fired with the new date.

### ⏩ Deprecate the `disableOpenPicker` prop

> The `disableOpenPicker` prop has been deprecated on all the picker components and will be removed in the next major release (v9.0.0).

Reviewer: ```suggestion
The `disableOpenPicker` prop has been deprecated on all Picker components and will be removed in the next major release (v9.0.0).
```

> If you only want to allow editing through the field, you can use the field component directly:

Reviewer: WDYT about additionally linking to the doc for anyone still unaware of the concept? 🤔 
```suggestion
If you only want to allow editing through the field, you can use the [field component](/x/react-date-pickers/fields/) directly:
```

```diff
-<DatePicker disableOpenPicker />
+<DateField />

-<TimePicker disableOpenPicker />
+<TimeField />

-<DateTimePicker disableOpenPicker />
+<DateTimeField />
```

```diff
-<DateRangePicker disableOpenPicker>
+<SingleInputDateRangeField> // If you want a single input for both dates.
+<MultiInputDateRangeField> // If you want one input for each date.

-<TimeRangePicker disableOpenPicker>
+<SingleInputTimeRangeField> // If you want a single input for both dates.
+<MultiInputTimeRangeField> // If you want one input for each date.

-<DateTimeRangePicker disableOpenPicker>
+<SingleInputDateTimeRangeField> // If you want a single input for both dates.
+<MultiInputDateTimeRangeField> // If you want one input for each date.
```

:::success

> Using the field instead of the picker will decrease significantly your bundle size since all the view component won't be bundled anymore.

Reviewer: Do you think it might be a common use case that Picker is only used for the manual field entry behavior in the whole application? 🤔 
IMHO, I'm not sure it's worth mentioning such a fact in the migration guide. 🤷 
```suggestion
Using the field instead of the picker significantly decreases your bundle size since all the view components won't be bundled anymore.
```

Author: My guess is that for people using this prop (a very small fraction of our users for sure), a good portion is doing it everywhere in their app.
But my main goal was to frame this change in a positive way.

Reviewer: Sure, the callout won't hurt. 👌 
Then let's keep only the typo suggestions as the open question. 👍
